x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/GeoPointFieldScript.java

```diff
@@ -49,20 +51,13 @@
         lookup,
         ctx
     ) {
+        final private GeoPoint scratch = new GeoPoint();
+
         @Override
         public void execute() {
             Object v = XContentMapValues.extractValue(field, leafSearchLookup.source().loadSourceIfNeeded());
-            if (v instanceof Map == false) {
-                return;
-            }
-            @SuppressWarnings("unchecked")
-            Map<String, ?> vs = (Map<String, ?>) v;
-            Object lat = vs.get("lat");
-            Object lon = vs.get("lon");
-            if (lat instanceof Number == false || lon instanceof Number == false) {
-                return;
-            }
-            emit(((Number) lat).doubleValue(), ((Number) lon).doubleValue());
+            GeoUtils.parseGeoPoint(v, scratch, true);
+            emit(scratch.lat(), scratch.lon());
```

Comment:
Which is the strategy for invalid formats? should an error be thrown or should be just catch it?

Reply:
I think we catch it in other places and don't emit values, what do you think @romseygeek ?

Reply:
Yes, elsewhere we catch exceptions and ignore

Reply:
I wonder if throwing or not an error should be controlled by something like the `ignored_flag`?. It might be surprising by users not to get any feedback why the point has not been emitted.

Reply:
I am a bit on the fence on this. It is true that users would like to know, on the other hand you would not want to break the query for one document that has some wrong format. This should be discussed for all runtime fields and eventually be streamlined. For now I would go for consistency and do what other runtime fields do, ignore errors.

Reply:
I think the current behaviour can bet the default one but we should have a way to get those exceptions. I am thinking of a user raising the question why a query with runtime fields is not working. We currently have no way to get this feedback as exceptions are swallow so it is going to be pretty tricky to understand the issue?

Reply:
yes agreed